Port is_set_to from TH/THC to ATen.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28597 Kill _th_zero binding, just use a simple native function instead.
* #28596 Kill THTensor_(match), which isn't used.
* #28595 Write out some set_ overloads instead of relying on code binding generation.
* #28514 Kill zero_dim_tensor_only codegen, it's not used anymore.
* #28513 Remove zero_dim_tensors only from _th_masked_fill_.
* #28512 Remove zero_dim_tensors only from _th_index_fill_.
* #28511 Kill _th_fill binding, which isn't used anymore.
* #28510 unfold: turn off device_guard
* #28475 Port TH/THC implementation of unfold to ATen.
* #28462 Fix scalar handling of unfold.
* #28435 Kill TH(C)Tensor_squeeze which isn't used anymore.
* **#28425 Port is_set_to from TH/THC to ATen.**
* #28422 is_set_to: unify TH/THC implmentation and genericize test_is_set_to.

Differential Revision: [D18063328](https://our.internmc.facebook.com/intern/diff/D18063328)

gchanan committed:
Port is_set_to from TH/THC to ATen.

[ghstack-poisoned]

mruberry commented on aten/src/ATen/native/TensorProperties.cpp:
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
   return result.copy_(self);
 }
+
+bool is_set_to(const Tensor &self, const Tensor& src) {
```
Tensor & vs Tensor&?

mruberry commented on aten/src/ATen/native/TensorProperties.cpp:
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
+      self.storage_offset() == src.storage_offset() &&
+      self.dim() == src.dim())
+  {
+    for (int64_t d = 0; d < self.dim(); ++d)
```
for (auto d = decltype(self.dim()){0}; d < ...

mruberry commented on aten/src/ATen/native/TensorProperties.cpp:
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
   auto result = at::empty_like(self, self.options(), memory_format);
   return result.copy_(self);
 }
+
```
Does this function need a comment? The name isn't super clear.

mruberry commented on aten/src/ATen/native/TensorProperties.cpp:
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
+  {
+    for (int64_t d = 0; d < self.dim(); ++d)
+    {
+      if (self.size(d) != src.size(d) || self.stride(d) != src.stride(d)) {
```
Your brace style is alternating between Allman and 1TBS (mandatory braces).

mruberry commented on aten/src/ATen/native/TensorProperties.cpp:
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
+    for (int64_t d = 0; d < self.dim(); ++d)
+    {
+      if (self.size(d) != src.size(d) || self.stride(d) != src.stride(d)) {
+        return true;
```
not return false here?

gchanan replied:
right.

mruberry reviewed (changes requested):
Need to verify return values. Style comments optional (but comments are always nice).

gchanan committed:
Update on "Port is_set_to from TH/THC to ATen."


Differential Revision: [D18063328](https://our.internmc.facebook.com/intern/diff/D18063328)

[ghstack-poisoned]

mruberry reviewed (approved):
Approving (assuming you'll fix lint).

gchanan committed:
Update on "Port is_set_to from TH/THC to ATen."


Differential Revision: [D18063328](https://our.internmc.facebook.com/intern/diff/D18063328)

[ghstack-poisoned]

gchanan committed:
Update on "Port is_set_to from TH/THC to ATen."


Differential Revision: [D18063328](https://our.internmc.facebook.com/intern/diff/D18063328)

[ghstack-poisoned]

gchanan committed:
Update on "Port is_set_to from TH/THC to ATen."


Differential Revision: [D18063328](https://our.internmc.facebook.com/intern/diff/D18063328)

[ghstack-poisoned]
